src/Ocelot/Requester/MessageInvokerHttpRequester.cs

```diff
@@ -0,0 +1,38 @@
+using Microsoft.AspNetCore.Http;
+using Ocelot.Logging;
+using Ocelot.Middleware;
+using Ocelot.Responses;
+
+namespace Ocelot.Requester;
+
+public class MessageInvokerHttpRequester : IHttpRequester
+{
+    private readonly IOcelotLogger _logger;
+    private readonly IExceptionToErrorMapper _mapper;
+    private readonly IMessageInvokerPool _messageHandlerPool;
+
+    public MessageInvokerHttpRequester(IOcelotLoggerFactory loggerFactory,
+        IMessageInvokerPool messageHandlerPool,
+        IExceptionToErrorMapper mapper)
+    {
+        _logger = loggerFactory.CreateLogger<MessageInvokerHttpRequester>();
+        _messageHandlerPool = messageHandlerPool;
+        _mapper = mapper;
+    }
+
+    public async Task<Response<HttpResponseMessage>> GetResponse(HttpContext httpContext)
+    {
+        var downstreamRequest = httpContext.Items.DownstreamRequest();
+        var messageInvoker = _messageHandlerPool.Get(httpContext.Items.DownstreamRoute());
+        try
+        {
+            var response = await messageInvoker.SendAsync(downstreamRequest.ToHttpRequestMessage(), httpContext.RequestAborted);
+            return new OkResponse<HttpResponseMessage>(response);
+        }
+        catch (Exception exception)
+        {
+            var error = _mapper.Map(exception);
```

Comment:
Ugly mappers again... 😢

Reply:
At the end the mappers are not that bad, but it's the way we are injecting them that is annoying.

Reply:
Not an issue for this PR for sure. As you know we have to do something with this design of Errors Mapper! Reason is "**wrong wrapping & overriding original status codes**". That's very bad!
Could you add some TODO sentence to these lines with a comment to pay attention lates please.
We have to return to this problem once again after Annual release...